lib/runtime/parseJson.ts

```diff
@@ -133,9 +133,6 @@ export function parseJsonString(s: string, pos: number): string | undefined {
             code += c.charCodeAt(0) - CODE_A + 10
           } else if (c >= "0" && c <= "9") {
             code += c.charCodeAt(0) - CODE_0
-          } else if (c === undefined) {
-            errorMessage("unexpected end")
-            return undefined
```

Comment:
I don't think it can be just removed, `c` can be undefined.

Reply:
yes and no. `s[pos]` can be undefined, but `s[pos]..toLowerCase()` can't because an undefined value will throw, and `toLowerCase` doesn't return undefined for any input.

Reply:
right, so this is a potential bug then - I don't remember if it is actually possible for pos to go out of range or was it simply defensive just in case - I think it can go out of range, in which case it needs to be fixed...

Reply:
Yeah, I was being lazy and not addressing the root cause, but figured that if it was possible people would be getting random TypeErrors and would have reported it? I can make the check accurate though.